sdk/cosmos/.azure-pipelines/cosmos.test.yml

```diff
@@ -26,27 +26,67 @@ jobs:
       vmImage: $(OSVmImage)
 
     steps:
+      - task: PowerShell@1
+        displayName: 'Download Public Cosmos Emulator'
+        inputs:
+          scriptType: inlineScript
+          inlineScript: |
+            Write-Host "Downloading Cosmos Emulator - $(EmulatorMsiUrl)"
+            wget "$(EmulatorMsiUrl)" -outfile "$env:temp\azure-cosmosdb-emulator.msi"
+            Write-Host "Finished Downloading Cosmos Emulator - $env:temp\azure-cosmosdb-emulator.msi"
+            dir "$env:temp"
+
+      - task: CmdLine@2
+        displayName: 'Cleanup already installed Azure Cosmos DB Emulator'
+        inputs:
+          script: |
+            echo "Deleting Azure Cosmos DB Emulator directory"
+            dir "%ProgramFiles%\"
+            rmdir /Q /S "%ProgramFiles%\Azure Cosmos DB Emulator"
+            echo "Directory after deleting"
+            dir "%ProgramFiles%\"
+
+      - task: CmdLine@2
+        displayName: 'Install Public Cosmos DB Emulator'
+        inputs:
+          script: |
+            choco install lessmsi
+            choco upgrade lessmsi
+            echo "Checking directory" 
+            dir "%ProgramFiles%"
+            mkdir "%TEMP%\Azure Cosmos DB Emulator"
+            lessmsi x "%TEMP%\azure-cosmosdb-emulator.msi" "%TEMP%\Azure Cosmos DB Emulator\"
+            dir "%TEMP%"
+            dir "%Temp%\Azure Cosmos DB Emulator\"
+
+      - task: PowerShell@1
+        displayName: 'Run Public Cosmos DB Emulator'
+        inputs:
+          scriptType: inlineScript
+          inlineScript: |
+            dir "$env:Temp\"
```

Comment:
Why are we doing a directory listing of all these directories?

Reply:
Same answer as above, this came from the original build on the cosmos side. Happy to remove everyone agrees.